proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -0,0 +1,321 @@
+* This approach seems likely to lead to trade-offs between optimal design and compliance with the modifier.
+* Being a source-compatible addition, modifiers can merely offer the *option* to be explicit.
+
+### Syntactic Matters First
+
+So far, existentials are the only built-in abstraction model (on par with generics and opaque types) that doesn't have its own idiosyncratic syntax; they are spelled as the bare protocol name or a composition thereof. Although the syntax strongly suggets that the protocol as a type and the protocol as a constraint are one thing, in practice, they serve different purposes, and this manifests most confusingly in the *"Protocol (the type) does not conform to Protocol (the constraint)"* paradox. This could be qualified as a missing feature in the language; the bottom line is that the syntax is tempting developers when it should be contributing to weighted decisions. Because using existential types is syntactically lightweight in comparison to using other abstractions, and similar to using a base class, the more possibilities they offer, the more *users are* vulnerable to unintended or inappropriate type erasure by following the path of initial least resistance.
+
+With regard to a source-compatible adoption of new adornments, we could
+* force all newly supported existential types to use the new syntax, i.e. `Any<P>` or [`any P` as a dual to `some P`](https://forums.swift.org/t/improving-the-ui-of-generics/22814#heading--clarifying-existentials), and
+* allow the respective protocols of all newly supported existential types to be annotated with an attribute to opt into the old syntax (to allow for source-compatible addition of requirements).
+
+Though, if we *were* going to introduce a new syntax for existentials, we think it'd be much less confusing if we took the potentially source-breaking path and did so uniformly, deprecating the existing syntax after a late-enough language version, than to have yet another attribute and two syntaxes where one only works some of the time. We also believe that drawing a tangible line between protocols that "do" and "do not" have limited access to their API is ill-advised due to the relative nature of this phenomenon.
+
+### Leave the Language in Its Existing State
+
+The current semantics regarding existential types prevent language users, especially library authors, from inheriting useful protocols due to the high cost of maintaining a type erasure construct. Namely, the high maintenance cost is a result of type erasure containers constituting hard-to-understand boilerplate code that entails thorough testing and a significant effort when expanding the protocol at hand. Furthermore, regular language users are often confused by the current restriction, and often lack the expertise or resources required for creating correct type-erasure contructs. 
+
```

Comment:
Since the debate in both threads was on the approach to be taken rather than the removal of the restriction, I think we are safe to leave the "do nothing" part out.

Reply:
I agree it's unnecessary, since it's, also, quite similar to Motivation.